api/repositories/service_binding_repository.go

```diff
@@ -272,6 +280,52 @@ func (r *ServiceBindingRepo) GetServiceBinding(ctx context.Context, authInfo aut
 	return serviceBindingToRecord(*serviceBinding), nil
 }
 
+func (r *ServiceBindingRepo) GetServiceBindingDetails(ctx context.Context, authInfo authorization.Info, bindingGUID string) (ServiceBindingDetailsRecord, error) {
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
```

Comment:
Getting the `CFServiceBinding` here is the same as getting it in [GetServiceBinding](https://github.com/cloudfoundry/korifi/blob/89d5234409cc06abb6462b21cd34cf6f2435212e/api/repositories/service_binding_repository.go#L256-L270). I would suggest extracting a reusable utility function to remove code duplication